src/lib/stepper/step-header.html

```diff
@@ -1,6 +1,10 @@
 <div class="mat-step-header-ripple" mat-ripple [matRippleTrigger]="_getHostElement()"></div>
-<div [class.mat-step-icon]="state !== 'number' || selected"
+<!-- @breaking-change 8.0.0 remove `mat-step-icon-not-touched` -->
+<div class="mat-step-icon"
```

Comment:
@arodr967 Hmm ok, it looks like maybe the touched class wasn't the issue after all. I'm still seeing screenshot diffs where the pencil is changed from blue to gray. Looking a little deeper I think the issue may be related to the `mat-step-icon` class. It looks like you changed it from conditionally applied to always applied and also changed the styles targeted at it to no longer set the background color. I'm assuming that's where these diffs are coming from. How do you feel about changing the classes as follows:

```html
<div class="mat-step-icon mat-step-icon-state-{{state}}"
     [class.mat-step-icon-selected]="selected"
     [ngSwitch]="state">
  ...
```

To me this seems nicer anyways, because each state will get its own class that users can target. If making the pencil gray by default was intentional, I can just add CSS in google to target the `-state-edit` class and make it blue again. And we can probably just remove the `-not-touched` class again since it seems it wasn't the cause of the problem. I can use the `-state-` classes and the `-selected` class to target the same elements anyways if needed.

Sorry for all the back and forth on this, sorting out the issues in google3 turned out to be more difficult than I anticipated

Reply:
Why would it not work for custom states? As long as they provide a string it should just insert it into the class name right?

Reply:
Yes but I wouldn't be able to predefine the classes since there could be many different types. Or would that be something that the developer could do on their side? Define the class? 🤔

Reply:
Simple way to fix the problem you're describing with the gray pencil is by just adding additional logic to *-selected* and *-not-selected* classes like so:

```
<div class="mat-step-icon"
     [class.mat-step-icon-selected]="selected || state == 'done' || state == 'edit'"
     [class.mat-step-icon-not-selected]="!selected && state != 'done' && state != 'edit'"
     [class.mat-step-icon-error]="state == 'error'"
     [ngSwitch]="state">
```

Reply:
I would rather change the classes to be like I suggested. It also makes the meaning of the `-selected` class more clear. I would expect the class to reflect the `selected` property, not all of these other conditions. I also think its useful for the user if we expose classes for their custom states

Reply:
Okay, I'll see what I can do in that case, thanks!

Reply:
Sorry took a while to get this done, been really busy! I have pushed the changes. Let me know what you think.
I had to make this `!important`:
```
.mat-step-icon-selected,
    .mat-step-icon-state-done,
    .mat-step-icon-state-edit {
      background-color: mat-color($primary) !important;
      color: mat-color($primary, default-contrast);
    }
```
Otherwise, the styles wouldn't get applied correctly!